### PR TITLE
remove MPI dependency when only using LCI parcelport

### PR DESCRIPTION
**Pull request by JiakunYan**
Previously the LCI parcelport always depends on MPI. This PR removes this dependency.

### CODE REVIEW

File: libs/core/lci_base/src/lci_environment.cpp [hunk 1]
```diff
@@ -221,11 +203,9 @@ namespace hpx { namespace util {
...
     std::string lci_environment::get_processor_name()
     {
-        char name[MPI_MAX_PROCESSOR_NAME + 1] = {'\0'};
-        int len = 0;
-        MPI_Get_processor_name(name, &len);
-
-        return name;
+        char hostname[HOST_NAME_MAX + 1] = "Unknown";
+        gethostname(hostname, HOST_NAME_MAX + 1);
+        return hostname;
```
Comment (hkaiser) [inline libs/core/lci_base/src/lci_environment.cpp:208]: Could you please use `asio::ip::host_name();` instead (needs `#include <asio/ip/host_name.hpp>`)?

**Review by hkaiser — CHANGES_REQUESTED** [review]

File: libs/core/lci_base/src/lci_environment.cpp [hunk 2]
```diff
@@ -221,11 +203,8 @@ namespace hpx { namespace util {
 
     std::string lci_environment::get_processor_name()
     {
-        char name[MPI_MAX_PROCESSOR_NAME + 1] = {'\0'};
-        int len = 0;
-        MPI_Get_processor_name(name, &len);
-
-        return name;
+        return asio::ip::host_name();
+        ;
```
Comment (hkaiser) [inline libs/core/lci_base/src/lci_environment.cpp:207]:
```suggestion
```

**Comment by JiakunYan** [comment]
Did a rebase on the current master branch.

**Review by hkaiser — APPROVED** [review]
LGTM, thanks!

**Comment by hkaiser** [comment]
bors merge

**Pull request merged**